providers/base/bin/grub_file_check.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+declare -A DICT_ARCH
+
+DICT_ARCH=( [x86_64]='x86_64' [aarch64]='arm64' [armhf]='armS' )
+FILE_CORE="/boot/grub/${DICT_ARCH[$(uname -m)]}-efi/core.efi"
+
+if [ -e "${FILE_CORE}" ]; then
+    echo "The file ${FILE_CORE} exists, shim and grub can be upgraded."
+    exit 0
+else
+    echo "Due to the absence of the file ${FILE_CORE}, upgrading shim and grub becomes impossible."
```

Comment:
If the system is using SecureBoot, more information should be shown, because it's a big issue for the end user.

You can check if the system is using SB using `mokutil --sb-state` (this should work on x86_64 and arm platforms), and if so, and the `core.efi` file is missing, you should add an extra warning.

This will be useful for people who review the submissions later on, to decide whether or not this is a blocking problem/regression that needs to be addressed.

Reply:
Following discussion with Kent, this part is not required anymore. Just make the job a cert-blocker.

Reply:
Done. Just marked it as cert-blocker.